compiler-rt/lib/sanitizer_common/sanitizer_allocator.cpp

```diff
@@ -59,7 +59,7 @@ static void *RawInternalAlloc(uptr size, InternalAllocatorCache *cache,
 
 static void *RawInternalRealloc(void *ptr, uptr size,
                                 InternalAllocatorCache *cache) {
-  uptr alignment = 8;
+  constexpr usize alignment = Max<usize>(8, sizeof(void *));
```

Comment:
Any reason for the switch in type here?

Reply:
It is the correct type, it's a size not a pointer/integer union (uinptr_t).

 This helps silence pedantic warnings for CHERI compilers since the value is a size_t (64-bit value) and not a uintptr_t (128 bit). No functional change for other architectures.